src/lib/currency/currency.ml

```diff
@@ -66,6 +66,31 @@ end = struct
   [%%define_locally
   Unsigned.(to_uint64, of_uint64, of_int, to_int, of_string, to_string)]
 
+  let precision = Unsigned.of_int 9
+
+  let to_formatted_string amount =
+    let whole = Unsigned.div amount precision in
+    let remainder = Unsigned.rem amount precision in
+    to_string whole ^ "." ^ to_string remainder
+
+  let of_formatted_string input =
+    let parts = String.split ~on:'.' input in
+    match parts with
+    | [whole] ->
+        of_string (whole ^ String.make (to_int precision) '0')
+    | [whole; decimal] ->
+        let decimal_length = String.length decimal in
+        let precision_int = to_int precision in
+        if
+          let open Int in
+          decimal_length > precision_int
+        then of_string (whole ^ String.sub decimal ~pos:0 ~len:precision_int)
```

Comment:
I think this and line 90 don't add the ".".
Any value to creating a helper that creates the final string out of the two parts?